scripts/docgen.lua

```diff
@@ -48,17 +49,25 @@ local function make_parts(fns)
 end
 
 local function make_section(indentlvl, sep, parts)
-  return indent(indentlvl, table.concat(make_parts(parts), sep))
+  local flat = make_parts(parts)
+  if not flat or #flat == 0 then
+    return ''
+  end
+  return indent(indentlvl, table.concat(flat, sep))
 end
 
 local function readfile(path)
   assert((vim.loop.fs_stat(path) or {}).type == 'file')
   return io.open(path):read '*a'
 end
 
-local lsp_section_template = [[
-## {{config_name}}
+local function relpath(from, to)
+  return to:gsub('^' .. vim.pesc(from) .. '/', '')
+end
 
+local lsp_section_template_txt = [[
+{{tagline}}
+==============================================================================
```

Comment:
help files put the line before the text, reverse of markdown
```suggestion
==============================================================================
{{tagline}}
```